[CI] SystemIndicesUpgradeIT class failing

**Build Scans:**
- [elasticsearch-periodic #7228 / 8.0.1_bwc](https://gradle-enterprise.elastic.co/s/ns6vi7qi2lsjy)
- [elasticsearch-periodic #7221 / 8.0.1_bwc](https://gradle-enterprise.elastic.co/s/pjzcqq4auerji)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.SystemIndicesUpgradeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.upgrades.SystemIndicesUpgradeIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [8.x] 2 failures in class org.elasticsearch.upgrades.SystemIndicesUpgradeIT (0.2% fail rate in 1000 executions)
- [8.x] 2 failures in step 8.0.1_bwc (100.0% fail rate in 2 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

@alexey-ivanov-es do you thing this is solved by your latest changes?
It seems to be failing across multiple backward compatbility and upgrade tests.
Looking at the logs, I see 
```
java.lang.AssertionError: expected empty context but was {X-elastic-product-origin=fleet}
```

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
@@ -23,6 +23,7 @@ import org.elasticsearch.logging.Logger;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.xcontent.json.JsonXContent;
 import org.elasticsearch.xpack.test.SecuritySettingsSourceField;
+import org.junit.BeforeClass;
 
 import java.io.IOException;
 import java.util.List;
@@ -39,6 +40,11 @@ public class SystemIndicesUpgradeIT extends AbstractUpgradeTestCase {
         "test_user",
         new SecureString(SecuritySettingsSourceField.TEST_PASSWORD)
     );
+
+    @BeforeClass
+    public static void avoidBugIn8_0_1() {
+        assumeTrue("https://github.com/elastic/elasticsearch/issues/125168", isOriginalClusterVersionAtLeast(Version.V_8_1_0));
+    }
 
     @Override
     protected Settings restAdminSettings() {

